vmm/src/vm.rs

```diff
@@ -635,16 +746,25 @@ impl<'a> Vm<'a> {
         fd.set_tss_address(arch::x86_64::layout::KVM_TSS_ADDRESS.raw_value() as usize)
             .map_err(Error::VmSetup)?;
 
-        // Create IRQ chip
-        fd.create_irq_chip().map_err(Error::VmSetup)?;
-
         // Supported CPUID
         let mut cpuid = kvm
             .get_supported_cpuid(MAX_KVM_CPUID_ENTRIES)
             .map_err(Error::VmSetup)?;
 
         let mut cpuid_patches = Vec::new();
+        let mut userspace_ioapic = false;
         if kvm.check_extension(Cap::TscDeadlineTimer) {
+            // Create split irqchip
+            // Only the local APIC is emulated in kernel, both PICs and IOAPIC
+            // are not.
+            let mut cap: kvm_enable_cap = Default::default();
+            cap.cap = KVM_CAP_SPLIT_IRQCHIP;
```

Comment:
Maybe this is just they way it is, but it feels a bit weird that enabling the split irqchip capability is conditional not on itself being available but just the TSC deadline timer. Should the check also include this too?

Reply:
Not sure I understand. This cap does not rely on any other cap. It's always available in the context of x86, but I put it under the TSC_DEADLINE_TIMER condition, because we need to be able to remove the PIT if we want to split the irqchip.

Reply:
Oh, so all Linux kernel versions have KVM_CAP_SPLIT_IRQCHIP when running on x86?

Reply:
How do we check that? The kvm_bindings that we have right now are 4.14 and 4.20 bindings, and they both have the KVM_CAP_SPLIT_IRQCHIP available.
But you're right, if we're running on an old kernel (< 4.14), how do we check availability for caps?

Reply:
(Incoming drive-by review) You can check extension on KVM_CAP_SPLIT_IRQCHIP.

Reply:
comment addressed!